cmd/swapd/main.go

```diff
@@ -411,6 +420,7 @@ func (d *daemon) make(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
+	// this closes the monero.WalletClient
```

Comment:
Yeah, converting the backend to use a has-a relationship for the monero and ethereum clients will make a lot of code more obvious. I thought about doing it in my current work, but I think it needs to be done in a PR dedicated to that change alone.